restli-server/src/main/java/com/linkedin/restli/server/symbol/RestLiSymbolTableRequestHandler.java

```diff
@@ -95,13 +97,22 @@ public boolean shouldHandle(Request request)
     // is in the form of /symbolTable or /symbolTable/<TABLENAME>
     //
     boolean isServiceScopedPath = request.getHeaders().containsKey(RestConstants.HEADER_SERVICE_SCOPED_PATH);
+    boolean isSymbolTableHeaderPresent = request.getHeaders().containsKey(SYMBOL_TABLE_REQUEST_HEADER);
+    int pathSize = pathSegments.size();
     if (isServiceScopedPath)
     {
-      return (pathSegments.size() == 3 && pathSegments.get(2).getPath().equals(SYMBOL_TABLE_URI_PATH));
+      return (pathSize == 3 && pathSegments.get(2).getPath().equals(SYMBOL_TABLE_URI_PATH));
+    }
+    else if (isSymbolTableHeaderPresent)
```

Comment:
I don't like the idea of adding another header for this. It will complicate the exchange protocol further. Is there some other way we can do this? Like say check routing to see if a resource with that name exists or not?

Or even better add a rule to disallow a rest.li resource from being named symbolTable.

@kbalasub wdyt?

Reply:
Having a reserved resource would make sense and seems most straightforward.

Reply:
@aman1309 can you please work with @sangm on the best path forward here?